spec/models/concerns/relatable_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe Relatable do
+  before(:all) do
+    @relatable_models = relatable_models
+    @countable_models = @relatable_models.map do |model|
+      model.counter_names
+    end.flatten.uniq
+  end
+
+  it 'makes sure the cached counters work' do
+    pp = create(:plant_population,
+                linkage_maps: create_list(:linkage_map, 3, plant_population: nil))
+    expect(pp.linkage_maps.count).to eq 3
+    expect(pp.reload.linkage_maps_count).to eq 3
+    pp.linkage_maps << create(:linkage_map, plant_population: nil)
+    expect(pp.linkage_maps.count).to eq 4
+    expect(pp.reload.linkage_maps_count).to eq 4
+    lms = create_list(:linkage_map, 2, plant_population: nil)
+    pp.linkage_maps = lms
+    expect(pp.reload.linkage_maps.count).to eq 2
```

Comment:
Intentional. This is how counter cache works - it does NOT work until you draw the model from the DB again.